lib/sqlalchemy/orm/events.py

```diff
@@ -2933,7 +3131,9 @@ def no_deleted(query, update_context):
 
         """
 
-    def before_compile_delete(self, query, delete_context):
+    def before_compile_delete(
+        self, query: Query[Any], delete_context: _O
```

Comment:
**mike bayer** (zzzeek) wrote:

delete_context here would be sqlalchemy.orm.query.BulkDelete

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4326